src/squidpy/gr/_niche.py

```diff
@@ -0,0 +1,432 @@
+        Optional if flavor == 'neighborhood'.
+    scale
+        If 'True', compute z-scores of neighborhood profiles.
+        Optional if flavor == 'neighborhood'.
+    abs_nhood
+        If 'True', calculate niches based on absolute neighborhood profile.
+        Optional if flavor == 'neighborhood'.
+    adj_subsets
+        List of adjacency matrices to use e.g. [1,2,3] for 1-hop,2-hop,3-hop neighbors respectively or "5" for 1-hop,...,5-hop neighbors. 0 (self) is always included.
+        Required if flavor == 'cellcharter'.
+    aggregation
+        How to aggregate count matrices. Either 'mean' or 'variance'.
+        Required if flavor == 'cellcharter'.
+    n_components
+        Number of components to use for GMM.
+        Required if flavor == 'cellcharter'.
+    random_state
+        Random state to use for GMM.
+        Optional if flavor == 'cellcharter'.
+    spatial_connectivities_key
+        Key in `adata.obsp` where spatial connectivities are stored.
+    """
+
+    # check whether anndata or spatialdata is provided and if spatialdata, check whether table_key is provided
+    if isinstance(adata, SpatialData):
+        if table_key is not None:
+            adata = adata.tables[table_key].copy()
+        else:
+            raise ValueError("Please specify which table to use with `table_key`.")
+    else:
+        adata = adata
+
+    if flavor == "neighborhood":
+        """adapted from https://github.com/immunitastx/monkeybread/blob/main/src/monkeybread/calc/_neighborhood_profile.py"""
+
+        # calculate the neighborhood profile for each cell (relative and absolute proportion of e.g. each cell type in the neighborhood)
+        rel_nhood_profile, abs_nhood_profile = _calculate_neighborhood_profile(
+            adata, groups, subset_groups, spatial_connectivities_key
+        )
+        # create AnnData object from neighborhood profile to perform scanpy functions
+        if not abs_nhood:
+            adata_neighborhood = ad.AnnData(X=rel_nhood_profile)
+        else:
+            adata_neighborhood = ad.AnnData(X=abs_nhood_profile)
+
+        # reason for scaling see https://monkeybread.readthedocs.io/en/latest/notebooks/tutorial.html#niche-analysis
+        if scale:
+            sc.pp.scale(adata_neighborhood, zero_center=True)
+
+        # mask obs to exclude cells for which no niche shall be assigned
+        if mask is not None:
+            mask = mask[mask.index.isin(adata_neighborhood.obs.index)]
+            adata_neighborhood = adata_neighborhood[mask]
+
+        # required for leiden clustering (note: no dim reduction performed in original implementation)
+        sc.pp.neighbors(adata_neighborhood, n_neighbors=n_neighbors, use_rep="X")
+
+        if resolutions is not None:
+            if not isinstance(resolutions, list):
+                resolutions = [resolutions]
+        else:
+            raise ValueError("Please provide resolutions for leiden clustering.")
```

Comment:
this could be `resolutions = [resolutions] if not isinstance(resolutions, list) else resolutions` but then you'd also have to validate the args for this specific niche flavour. Using spearate functions, these checks could be done before